app/addons/dataimporter/components.react.jsx

```diff
@@ -0,0 +1,797 @@
+        </div>
+      );
+    },
+
+    boxIsDraggingOver: function () {
+      return (
+        <div className="dropzone dragging-file-in-background"
+          onDragOver={this.dragOver}
+          onDragLeave={this.endDragOver}
+          onDrop={this.drop}>
+          <div className="dropzone-msg draggingover">
+            <span className="fonticon icon-file-text-alt">
+            </span>
+            <span className="loading-msg">
+              Drop your file.
+            </span>
+          </div>
+        </div>
+      );
+    },
+
+    boxIsLoading: function () {
+      var loadingBoxBigFileMessage = this.props.isBigFile ?
+        this.loadingBoxBigFileMessage() : '';
+
+      return (
+        <div className={"dropzone loading-background"}>
+          <div className="dropzone-msg loading">
+            Loading...
+            {loadingBoxBigFileMessage}
+            <Components.LoadLines />
+          </div>
+        </div>
+      );
+    },
+
+    boxShowLimitInfo: function () {
+      return (
+         <div className="dropzone limit-info"
+          onDragOver={this.dragOver}
+          onDragLeave={this.endDragOver}
+          onDrop={this.drop}>
+          <div className="dropzone-msg">
+            <p>150 MB filesize limit</p>
+            <p>Only .csv files will import correctly</p>
+            <p>Fine grained import options are only for files under 200KB</p>
+          </div>
+          {this.fileLimitLink("Close")}
+        </div>
+      );
+    },
+
+    render: function () {
+      var box = this.defaultBox();
+
+      if (this.state.draggingOver) {
+        box = this.boxIsDraggingOver();
+      } else if (this.state.loading) {
+        box = this.boxIsLoading();
+      } else if (this.state.showLimitInfo) {
+        box = this.boxShowLimitInfo();
+      }
+      return box;
+    }
+  });
+
+  var DataImporterPreviewData = React.createClass({
+    getInitialState: function () {
+      return {
+        left: 0
+      };
+    },
+    bigFilePreviewWarning: function () {
+      var rowShown = this.props.rowShown,
+          totalRows = this.props.rowsTotal;
+
+      return (
+        <div className="top-row">
+          <div className="big-file-info-message">
+            <p className="big-file-preview-limit-info-message">
+              Because of the size of this file, this preview only shows the
+              first {rowShown} rows, out of {totalRows} rows total.
+              However, if you choose to load the data into a database, the
+              entirety of the file (all {totalRows} rows) will be imported.
+            </p>
+          </div>
+          {this.loadingIntoDB()}
+        </div>
+      );
+    },
+    componentDidMount: function () {
+      document.getElementById('preview-page')
+        .addEventListener('scroll', this.handleScroll);
+    },
+    componentWillUnmount: function () {
+      document.getElementById('preview-page')
+        .removeEventListener('scroll', this.handleScroll);
+    },
```

Comment:
I think if you move the error message into the `DataImporterController` or another parent element you don't have to calculate the left value all the time which is costing a lot of performance as it triggers a repaint in the browser on every pixel that is scrolled

https://gist.github.com/paulirish/5d52fb081b3570c81e3a